mongomock/store.py

```diff
@@ -1,11 +1,91 @@
 import collections
 import datetime
+from contextlib import contextmanager
+
 import mongomock  # Used for utcnow - please see https://github.com/mongomock/mongomock#utcnow
 import six
 import six.moves
 import threading
 
-lock = threading.RLock()
+
+class RWLock:
```

Comment:
Please move the thread specific classes to a mongomock/thread module

Reply:
Done